identity/manager.go

```diff
@@ -95,13 +100,99 @@ func (m *Manager) Create(ctx context.Context, i *Identity, opts ...ManagerOption
 	}
 
 	if err := m.r.PrivilegedIdentityPool().CreateIdentity(ctx, i); err != nil {
+		if errors.Is(err, sqlcon.ErrUniqueViolation) {
+			return m.generateCredentialHint(ctx, err, i)
+		}
 		return err
 	}
 
 	trace.SpanFromContext(ctx).AddEvent(events.NewIdentityCreated(ctx, i.ID))
 	return nil
 }
 
+func (m *Manager) generateCredentialHint(ctx context.Context, e error, i *Identity) (err error) {
+	var hints []string
+	var message string
+	for _, va := range i.VerifiableAddresses {
+		conflictingAddress, err := m.r.PrivilegedIdentityPool().FindVerifiableAddressByValue(ctx, va.Via, va.Value)
+		if errors.Is(err, sqlcon.ErrNoRows) {
+			continue
+		}
+		if err != nil {
+			m.r.Logger().WithError(err).Error("Failed to generate credential hint")
+			continue
+		}
+		message = fmt.Sprintf("An account for %s already exists.", va.Value)
+		hintIdentity, err := m.r.PrivilegedIdentityPool().GetIdentity(ctx, conflictingAddress.IdentityID, ExpandCredentials)
```

Comment:
ah yeah, you're already querying the credentials here, so I guess move this up and query by the identifier.